### ISSUE TITLE
different results on different machine

### ISSUE THREAD

**Issue by jiayeah9508** [tensorflow/tensorflow#28390]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): win10(10.0 17134  10.0 15063)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: unclear
- TensorFlow installed from (source or binary): anaconda navigator
- TensorFlow version (use command below): 1.13.1
- Python version: 3.7.0
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source): spyder 3.3.1
- CUDA/cuDNN version: CPU
- GPU model and memory: CPU

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
different results on different machine
**Describe the expected behavior**
same results on different machine
**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
import numpy as np
import csv
import matplotlib.pyplot as plt
import tensorflow as tf
from sklearn.utils import shuffle
from batch import Dataset

def generate(sample_size,mean,cov,diff,regression):
    num_classes = 2
    samples_per_class = int(sample_size/2)
    np.random.seed(1)
    X0 = np.random.multivariate_normal(mean,cov,samples_per_class)
    Y0 = np.zeros(samples_per_class)
    
    for ci,d in enumerate(diff):
        X1 = np.random.multivariate_normal(mean+d,cov,samples_per_class)
        Y1 = (ci+1)*np.ones(samples_per_class)
        
        X0 = np.concatenate((X0,X1))
        Y0 = np.concatenate((Y0,Y1))
    
    if regression==False:
        class_ind = [Y0==class_number for class_number in range(num_classes)]
        Y0 = np.asarray(np.hstack(class_ind),dtype=np.float32)
        
    X,Y = shuffle(X0,Y0,random_state=1)
    
    return X,Y

np.random.seed(10)
num_classes = 2
mean = np.random.randn(num_classes)
cov = np.eye(num_classes)
X,Y = generate(100,mean,cov,[3.0],True)
colors = ['r' if l==0 else 'b' for l in Y[:]]
plt.scatter(X[:,0],X[:,1],c=colors)
plt.xlabel('Scaled age (in yrs)')
plt.ylabel('Tumor size (in cm)')
plt.show()
lab_dim =1

path = 'F:\\实验室代码\\python\\autoencoder\\testData\\线性逻辑回归训练集.csv'
f = open(path ,'w',newline='',encoding='utf-8') 
writer = csv.writer(f)
for item in X :
   writer.writerow(item)  
f.close()

input_dim = X.shape[1]
lab_dim = 1
input_features = tf.placeholder(tf.float32,[None,input_dim])
input_labels = tf.placeholder(tf.float32,[None,lab_dim])
W = tf.Variable(tf.random_normal([input_dim,lab_dim],seed=1),name='weight')
b = tf.Variable(tf.zeros([lab_dim]),name='bias')

output = tf.nn.sigmoid(tf.matmul(input_features,W)+b)
cross_entropy = -(input_labels*tf.log(output)+(1-input_labels)*tf.log(1-output))
ser = tf.square(input_labels-output)

loss = tf.reduce_mean(cross_entropy)
err = tf.reduce_mean(ser)
optimizer = tf.train.AdamOptimizer(0.04)
train = optimizer.minimize(loss)

maxEpochs = 51
minibatchSize = 25

path = 'F:\\实验室代码\\python\\autoencoder\\testData\\线性逻辑回归优化前权重.csv'
f_1 = open(path ,'w',newline='',encoding='utf-8') 
writer_1 = csv.writer(f_1)

path = 'F:\\实验室代码\\python\\autoencoder\\testData\\线性逻辑回归优化后权重.csv'
f_2 = open(path ,'w',newline='',encoding='utf-8') 
writer_2 = csv.writer(f_2)

path = 'F:\\实验室代码\\python\\autoencoder\\testData\\所有epoch的index.csv'
f_3 = open(path ,'w',newline='',encoding='utf-8') 
writer_3 = csv.writer(f_3)

with tf.Session() as sess:
    sess.run(tf.global_variables_initializer())
    
    w_1 = sess.run(W)  #优化前的权重
    for item in w_1 :
        writer_1.writerow(item)  
    f_1.close()

    index_all_epoch = []  #所有epoch
    for epoch in range(maxEpochs):
        sumerr=0
        index = Dataset(np.arange(0, 100))  #index为下标数组，从0到n_examples-1的整数
        index_whole = np.array([]) #一次epoch 
        seed = epoch
        for i in range(np.int32(len(Y)/minibatchSize)):
            
            tempindex = index.next_batch(minibatchSize,seed)  #将下标数组打乱，取出前面的下标
            
            index_whole = np.append(index_whole,tempindex)
            
            x1 = X[tempindex]
            y1 = np.reshape(Y[tempindex],[-1,1])
            tf.reshape(y1,[-1,1])
            _,lossval,outputval,errval = sess.run([train,loss,output,err],feed_dict = {input_features:x1,input_labels:y1})
            sumerr = sumerr + errval
        print('Epoch:','%04d' %(epoch+1),'cost=','{:.9f}'.format(lossval),'err=',sumerr/np.int32(len(Y)/minibatchSize))
        index_all_epoch.append(index_whole)
    
    w_2 = sess.run(W)  #优化后的权重
    for item in w_2 :
        writer_2.writerow(item)  
    f_2.close()
    
    for item in index_all_epoch:#所有epoch
        writer_3.writerow(item)
    f_3.close()
        
        

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by gadagashwini** [comment]
@jiayeah9508 please provide a minimum code snippet to reproduce the issue reported here. Thanks!

**Comment by gadagashwini** [comment]
Closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
